src/main/java/org/openrewrite/java/testing/jmockit/ArgumentMatchersRewriter.java

```diff
@@ -100,16 +101,24 @@ private J.MethodInvocation rewriteMethodInvocation(J.MethodInvocation invocation
             invocation = invocation.withSelect(rewriteMethodInvocation((J.MethodInvocation) invocation.getSelect()));
         }
         // in mockito, argument matchers must be used for all arguments or none
-        boolean hasArgumentMatcher = false;
-        List<Expression> arguments = invocation.getArguments();
-        for (Expression methodArgument : arguments) {
-            if (isJmockitArgumentMatcher(methodArgument)) {
-                hasArgumentMatcher = true;
-                break;
+        List<Expression> arguments = new ArrayList<>(invocation.getArguments().size());
```

Comment:
Could simplify this to:
       List<Expression> arguments = invocation.getArguments();
       // replace this.matcher with matcher, otherwise it's ignored
       arguments.replaceAll(arg -> { 
               if (methodArgument instanceof J.FieldAccess) {
                    J.FieldAccess fieldAccess = (J.FieldAccess) methodArgument;
                     if (fieldAccess.getTarget() instanceof J.Identifier &&
                            ((J.Identifier) fieldAccess.getTarget()).getSimpleName().equals("this")) {
                               return fieldAccess.getName();
                    }
               }
               return arg;
      });

Reply:
Personally I prefer the "functional programming" way, but I just followed the "traditional" ways in this file.
After a rough search I found that there are already plenty of code using functional programming, so I take your suggestion